### ISSUE TITLE
Binance Future: create_order() with custom params (leverage, marginMode, closePosition, HedgeMode, reduceOnly, etc.)

### ISSUE THREAD

**Issue by ilpomo** [ccxt/ccxt#19086]
### Operating System

Fedora Linux 38

### Programming Languages

Python

### CCXT Version

4.0.80

### Description

After looking for a solution, I think the official documentation should be drastically improved by **explicitly** explain and show with examples how to setup correctly a CCXT exchange instance to submit different types of orders.

Don't get me wrong, you can (probably) find a solution online most of the time, but you end up switching between the official CCXT documentation, this repository issues, StackOverflow and then "waste" quite some time mixing the different solutions you found in quite dumb/frustrating trial and error phase before achieving want you want.

Imho the documentation should offer an exchange specific section where it is explicitly showed how to correctly submit different types of orders with exchange specific custom params passed as argument. And it should be kept updated.

In my particular case, can someone clarify me how to:
- Correctly instaciate a Binance USD-M Futures client to create orders for the perpetual markets (for example BTC/USDT:USDT).
- Create market / limit / stop-loss / take-profit orders with custom params argument? For example, custom `leverage`, `marginMode`, `closePosition`, `HedgeMode`, `reduceOnly`, etc.
- How to use (or not) `HedgeMode`? If a open a new buy/long (or sell/short) position and then create an order in opposite direction using the same amount, instead of opening a new position (`HedgeMode=False`) it closes the previously opened one (`HedgeMode=True`).
- How can I pass `closePosition=True` in params while indicating `undefined` (`None` in Python?) (as suggested by @kroitor many times in multiple issues)

Btw, below some example code I made for testing different types of order with custom params. Thank you.

### Code

    from ccxt import binanceusdm

    if __name__ == "__main__":

        client = binanceusdm({
            "apiKey": "MY_API_KEY",
            "secret": "MY_SECRET_KEY"
            }
        )
    
        binance.options["defaultType"] = "future"  # why is it necessary since I have already imported binanceusdm instead of binance?
        binance.options["adjustForTimeDifference"] = True  # should not it be enabled by default anyway?
    
        BASE = "XRP"
        QUOTE = "USDT"
        symbol = f"{BASE}/{QUOTE}:{QUOTE}"
        
        size = 2 / 100  # 2% of free USDT to invest in new order
        leverage = 5  # 5x leverage

        response = client.fapiprivate_post_leverage({"symbol": "XRPUSDT", "leverage": leverage})
        balance = client.fetch_balance()
        positions = client.fetch_positions()

        free = balance[QUOTE]["free"]
        price = binance.fetch_last_prices(symbols=[symbol,])[symbol]["price"]

        amount = (free * size * leverage) / price * 0.99  # 0.99 to ensure the order will be filled
        
        long = client.create_order(
        symbol=symbol,
        type="market",
        side="buy",
        amount=amount,
        price=price,
        params={
            "leverage": leverage,  <- ignored, in Binance app orders will always show up having 1x leverage
            "marginType": "CROSSED"
            }
        )
        
        short = client.create_order(
        symbol=symbol,
        type="market",
        side="sell",
        amount=amount,
        price=price,
        params={
            "leverage": leverage,  <- ignored, in Binance app orders will always show up having 1x leverage
            "marginType": "CROSSED"
            }
        )
        
        # stop loss?
        
        stop_loss = client.create_order(
        symbol=symbol,
        type="market",
        side="sell",  # "sell" for long or "buy" for short, but anyway, why it does't open a new position instead of closing them?
        amount=None,  # <- can't be None
        price=None,  # <- can't be None
        params={
            "closePosition": True,
            "marginType": "CROSSED"
            }
        )
       
        # take profit?

**Comment by Dan-krm** [comment]
Hey @ilpomo thanks for the recommendations. I don't think there is any plans to add this exchange specific information to the documentation at the moment because we've been working on making the documentation a bit more concise and that might make it pretty bloated and we can usually abstract away most exchange specific params into CCXT's unified methods. There is an examples section of the repo https://github.com/ccxt/ccxt/tree/master/examples/py if there is any questions you have about a specific implementation you can open an issue and we can create a new example.

For closing a position you can use createOrder with the reduceOnly param set to true, there is also some newly added unified methods called `closePosition` and `closeAllPositions`: https://docs.ccxt.com/#/README?id=closing-positions

For some of your other questions:
- You can use different market types by setting the type param or option to 'spot', 'swap', 'future', and 'option' and you can set the subType to 'linear' or 'inverse' `BTC/USDT:USDT` is a linear swap market, methods like createOrder that require the symbol derive this information for you from the symbol so you don't need to fill out any type or subType parameters. If you want to use spot margin you just use a spot market like BTC/USDT or set the type to spot while also setting the marginMode to 'isolated' or 'cross'
- Generally all of the methods in CCXT have a see section in the docstring that links to the endpoint being used, you can review the endpoint being used in the exchange specific API documentation to get information on exchange specific params and their data types.
- Hedge mode in Binance allows traders to simultaneously hold both long (buy) and short (sell) positions on the same trading pair. https://www.binance.com/en/support/faq/what-is-hedge-mode-and-how-to-use-it-360041513552
- For the closePosition question you can view the exchange specific params in the exchange API documentation or you can use one of the new unified closePosition or closeAllPositions methods.
